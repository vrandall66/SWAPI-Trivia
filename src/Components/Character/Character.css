.Character {
  background-color: black;
  margin-bottom: 30px;
  width: 250px;
}

.favoriteIcon {
  width: 35px;
  margin-top: 10px;
}