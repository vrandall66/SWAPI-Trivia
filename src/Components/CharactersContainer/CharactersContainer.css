.CharactersContainer {
  color: #ffe81f;
  display: flex;
  flex-direction: column;
}

.CharactersContainer-h1 {
  height: 25vh;
  overflow: hidden;
  text-transform: uppercase;
}

.CharactersContainer-section {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  height: 70vh;
  overflow: scroll;
  width: 100%;
}

.BB8--Parent-Container {
  height: 100%;
  width: 100%;
}

.BB8--loading {
  height: 100%;
  margin-left: 50%;
  margin-top: 5%;
}

.BB8 {
  display: flex;
}

.MoviesContainer-loading {
  height: 350px;
  margin-top: 5%;
  width: 500px;
}

.episodeTitle {
  color: #ffe81f;
  margin: 0;
  font-family: "Roboto Condensed", sans-serif;
  font-size: 40px;
  margin-top: 20px;
}
