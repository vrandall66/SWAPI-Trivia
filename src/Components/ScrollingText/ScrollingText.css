.body {
  display: flex;
  align-content: center;
  width: 100%;
  height: 20%;
  justify-content: center;
  overflow: hidden;
  margin-top: 43px;
}

.scrolling-Text {
  align-items: center;
  display: flex;
  justify-content: center;
  height: 150px;
  perspective: 200px;
  color: #FFE81F;
  font-family: "Pathway Gothic One", sans-serif;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 300%;
  text-align: justify;
  width: 100%;
}

.scrollingText--Episode-info {
  justify-content: center;
}

h1 {
  text-transform: uppercase;
}

.crawl {
  position: relative;
  text-transform: uppercase;
  transform-origin: 50% 100%;
  animation: crawl 400s linear;
  width: 250%;
}

@keyframes crawl {
  0% {
    top: 200px;
    transform: rotateX(20deg) translateZ(120px);
  }
  100% {
    top: -6000px;
    transform: rotateX(60deg) translateZ(-2500px);
  }
}
