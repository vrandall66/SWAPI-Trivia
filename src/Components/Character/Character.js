import React from "react";
import "./Character.css";
import "../CharactersContainer/CharactersContainer.css"
import notFavorited from '../../images/not-favorite-ds.svg'
import favorited from '../../images/favorited-ds.svg'

const Character = ({ characterInfo, updateCharacterFavorite }) => {
  return characterInfo.map( (character) => {
    const { name, species, homeworld, films, characterid, favorite } = character
    const favoriteCharacter = character => {
      let {
        name,
        species,
        homeworld,
        films,
        characterid,
        favorite
      } = character;
      favorite = !favorite;
      return { name, species, homeworld, films, characterid, favorite };
    };
    return (
      <div className="Character" key={characterid}>
        <input 
          className='favoriteIcon'
          type='image' 
          alt='favoriteIcon' 
          src={favorite ? favorited : notFavorited} 
          onClick={() => updateCharacterFavorite(favoriteCharacter(character))}
        />
        <h1>{name}</h1>
        <p>{species}</p>
        <p>{homeworld.homeName}</p>
        <p>{homeworld.homePopulation}</p>
        <p>{films}</p>
      </div>
    );
  });
};

export default Character;
