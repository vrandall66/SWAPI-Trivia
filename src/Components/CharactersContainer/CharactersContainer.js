import React from "react";
import Character from "../Character/Character";
import ScrollingText from "../ScrollingText/ScrollingText";
import BB8Loading from "../../../src/images/BB8Loading.gif";
import "./CharactersContainer.css";
import { fetchCharacter } from "../../ApiCalls/apiCalls";

class CharactersContainer extends React.Component {
  constructor({ episode, addFavoriteCharacter, checkFavoriteStatus }) {
    super();
    this.addFavoriteCharacter = addFavoriteCharacter;
    this.checkFavoriteStatus = checkFavoriteStatus;
    this.state = {
      episode,
      characters: []
    };
  }

  componentDidMount = () => {
    const fetchCharacters = this.state.episode.characters.map(character => {
      return fetchCharacter(character);
    });
    return Promise.all(fetchCharacters).then(characters =>
      this.createCharacterCard(characters)
    );
  };

  createCharacterCard = characters => {
    const characterCards = characters.map(character => {
      return {
        name: character.name,
        species: character.response[0],
        homeworld: character.response[1],
        films: character.response[2],
        characterid: parseInt(character.url.split('/').splice(5,1).pop()),
        favorite: this.checkFavoriteStatus(parseInt(character.url.split('/').splice(5, 1).pop()))
      };
    });
    this.setState({ characters: characterCards });
  };

  updateCharacterFavorite = favoriteCharacter => {
    const updatedCharacters = this.state.characters;
    const characterIndex = updatedCharacters
      .map(character => character.characterid)
      .indexOf(favoriteCharacter.characterid);
    updatedCharacters.splice(characterIndex, 1, favoriteCharacter);
    this.setState({ characters: updatedCharacters });
    this.addFavoriteCharacter(favoriteCharacter);
  };

  render() {
    return (
      <div className="CharactersContainer">
        <ScrollingText episode={this.state.episode} />
        <section className="CharactersContainer-section">
          {this.state.characters.length === 0 ? (
            <div className="BB8--Parent-Container">
              <div className="BB8">
                <div className="BB8--loading">
                  <h1>Loading Characters</h1>
                  <img
                    src={BB8Loading}
                    alt="Loading GIF"
                    className="MoviesContainer-loading"
                  />
                </div>
              </div>
            </div>
          ) : (
            <Character
              characterInfo={this.state.characters}
              updateCharacterFavorite={this.updateCharacterFavorite}
            />
          )}
        </section>
      </div>
    );
  }
}

export default CharactersContainer;
