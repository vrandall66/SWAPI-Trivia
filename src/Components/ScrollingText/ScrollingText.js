import React from "react";
import "./ScrollingText.css";

const ScrollingText = ({ episode }) => {
  return (
    <div className="body">
      <section className="scrolling-Text">
        <div className="crawl">
          <div className="scrollingText--Episode-info">
            <h1>
              Episode {episode.episode_id} {episode.title}
            </h1>
          </div>
          <h1>{episode.opening_crawl}</h1>
        </div>
      </section>
    </div>
  );
};

export default ScrollingText;
