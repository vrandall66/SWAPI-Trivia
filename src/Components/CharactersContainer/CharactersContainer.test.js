import React from "react";
import { shallow } from "enzyme";
import CharactersContainer from "./CharactersContainer";
import ScrollingText from "../ScrollingText/ScrollingText";
import Character from "../Character/Character";

describe("CharactersContainer", () => {
  let wrapper1;
  let wrapper2;
  let BB8Loading;
  let episode = {
    characters: [],
    created: "",
    director: "",
    edited: "",
    episode_id: 0,
    opening_crawl: "",
    planets: [],
    producer: "",
    releaseDate: "",
    species: [],
    starships: [],
    title: "The Phantom Menace",
    url: "",
    vehicles: []
  };
  let addFavoriteCharacter = jest.fn();

  beforeEach(() => {
    wrapper1 = shallow(
      <CharactersContainer
        episode={episode}
        addFavoriteCharacter={addFavoriteCharacter}
      />
    );
  });

  wrapper2 = shallow(
    <div className="CharactersContainer">
      <ScrollingText episode={episode} />
      <section className="CharactersContainer-section">
        {episode.characters.length === 0 ? (
          <div className="BB8--Parent-Container">
            <div className="BB8">
              <div className="BB8--loading">
                <h1>Loading Characters</h1>
                <img
                  src={BB8Loading}
                  alt="Loading GIF"
                  className="MoviesContainer-loading"
                />
              </div>
            </div>
          </div>
        ) : (
          <Character
            characterInfo={this.state.characters}
            updateCharacterFavorite={this.updateCharacterFavorite}
          />
        )}
      </section>
    </div>
  );

  it("should match the snapshot when menu is collapsed", () => {
    expect(wrapper1).toMatchSnapshot();
  });

  it("should match the snapshot when menu is open", () => {
    expect(wrapper2).toMatchSnapshot();
  });

  it("should update character favorites", () => {
    const mockState = {
      episode,
      characters: [
        {
          name: "Yoda",
          species: "Yoda's species",
          homeworld: {
            homeName: "unknown",
            homePopulation: "unknown"
          },
          films: [
            "The Empire Strikes Back",
            "Attack of the Clones",
            "The Phantom Menace",
            "Revenge of the Sith",
            "Return of the Jedi"
          ],
          characterid: 20,
          favorite: true
        },
        {
          name: "R2-D2",
          species: "Droid",
          homeworld: {
            homeName: "Naboo",
            homePopulation: 4500000000
          },
          films: [
            "The Empire Strikes Back",
            "Attack of the Clones",
            "The Phantom Menace",
            "Revenge of the Sith",
            "Return of the Jedi",
            "A New Hope",
            "The Force Awakens"
          ],
          characterid: 3,
          favorite: true
        },
        {
          name: "Leia Organa",
          species: "Human",
          homeworld: {
            homeName: "Alderaan",
            homePopulation: 2000000000
          },
          films: [
            "The Empire Strikes Back",
            "Revenge of the Sith",
            "Return of the Jedi",
            "A New Hope",
            "The Force Awakens"
          ],
          characterid: 5,
          favorite: true
        }
      ],
      remainingCharacters: []
    };

    let newUnFavorite = {
      characterid: 5,
      favorite: false,
      name: "Leia Organa",
      films: [
        "The Empire Strikes Back",
        "Revenge of the Sith",
        "Return of the Jedi",
        "A New Hope",
        "The Force Awakens"
      ],
      species: "Human",
      homeworld: {
        homeName: "Alderaan",
        homePopulation: 2000000000
      }
    };

    wrapper1.instance().setState(mockState);

    wrapper1.instance().updateCharacterFavorite(newUnFavorite);

    expect(wrapper1.state().characters).toEqual([
      {
        characterid: 20,
        favorite: true,
        films: [
          "The Empire Strikes Back",
          "Attack of the Clones",
          "The Phantom Menace",
          "Revenge of the Sith",
          "Return of the Jedi"
        ],
        homeworld: {
          homeName: "unknown",
          homePopulation: "unknown"
        },
        name: "Yoda",
        species: "Yoda's species"
      },
      {
        characterid: 3,
        favorite: true,
        films: [
          "The Empire Strikes Back",
          "Attack of the Clones",
          "The Phantom Menace",
          "Revenge of the Sith",
          "Return of the Jedi",
          "A New Hope",
          "The Force Awakens"
        ],
        homeworld: {
          homeName: "Naboo",
          homePopulation: 4500000000
        },
        name: "R2-D2",
        species: "Droid"
      },
      {
        characterid: 5,
        favorite: false,
        films: [
          "The Empire Strikes Back",
          "Revenge of the Sith",
          "Return of the Jedi",
          "A New Hope",
          "The Force Awakens"
        ],
        homeworld: {
          homeName: "Alderaan",
          homePopulation: 2000000000
        },
        name: "Leia Organa",
        species: "Human"
      }
    ]);
  });
});
