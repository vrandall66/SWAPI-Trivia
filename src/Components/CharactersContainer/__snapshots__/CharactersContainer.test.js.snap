// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersContainer should match the snapshot when menu is collapsed 1`] = `
<div
  className="CharactersContainer"
>
  <ScrollingText
    episode={
      Object {
        "characters": Array [],
        "created": "",
        "director": "",
        "edited": "",
        "episode_id": 0,
        "opening_crawl": "",
        "planets": Array [],
        "producer": "",
        "releaseDate": "",
        "species": Array [],
        "starships": Array [],
        "title": "The Phantom Menace",
        "url": "",
        "vehicles": Array [],
      }
    }
  />
  <section
    className="CharactersContainer-section"
  >
    <div
      className="BB8--Parent-Container"
    >
      <div
        className="BB8"
      >
        <div
          className="BB8--loading"
        >
          <h1>
            Loading Characters
          </h1>
          <img
            alt="Loading GIF"
            className="MoviesContainer-loading"
            src="BB8Loading.gif"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`CharactersContainer should match the snapshot when menu is open 1`] = `
<div
  className="CharactersContainer"
>
  <ScrollingText
    episode={
      Object {
        "characters": Array [],
        "created": "",
        "director": "",
        "edited": "",
        "episode_id": 0,
        "opening_crawl": "",
        "planets": Array [],
        "producer": "",
        "releaseDate": "",
        "species": Array [],
        "starships": Array [],
        "title": "The Phantom Menace",
        "url": "",
        "vehicles": Array [],
      }
    }
  />
  <section
    className="CharactersContainer-section"
  >
    <div
      className="BB8--Parent-Container"
    >
      <div
        className="BB8"
      >
        <div
          className="BB8--loading"
        >
          <h1>
            Loading Characters
          </h1>
          <img
            alt="Loading GIF"
            className="MoviesContainer-loading"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;
